src/Serializer/ContextTrait.php

```diff
@@ -30,4 +35,34 @@ private function initContext(string $resourceClass, array $context): array
             'resource_class' => $resourceClass,
         ]);
     }
+
+    private function addRequestContext(Request $request, array $context): array
+    {
+        return array_merge($context, [
```

Comment:
Why  do we need to add so much data to the context? We're creating an indirect coupling to HTTP here, and the whole point of this change is to decouple from HTTP.

Reply:
Actually, HTTP-related data must not leak outside of `ReadListener`. Filters are already extracted in this class, there are no needs for these new keys.

Reply:
Please see the `GroupFilter` and the `PropertyFilter`. These filters are very coupled to the request data.
The `SerializerContextFactory` needs also a lot of request data in order to build the serializer context (related to `api_allow_update`, to CSV, ...).
It also adds the `request_uri` key used in some serializers.
I don't know how you would like to change all this.

Reply:
The `request_stack` should be injected directly in those specific filters. But copying the full HTTP request in the context isn't a good idea.

Reply:
I can do it for the filters, but not for the other cases.

Reply:
Also, I think a direct coupling with the request stack is not really a good idea, especially if we want to be able to manage other frameworks than Symfony.

Reply:
Another thing I thought: a lot of users **need** to have the request context for creating their own context factory. You can take this question in SO as an example: https://stackoverflow.com/questions/60340261/how-to-set-normalization-groups-based-on-users-role-and-request-method/60346166#60346166